### PR TITLE
Fixing 'UTC' did not equal 'Etc/UTC' failure on Mac

### LINKED ISSUE #7505: Unit tests fail on Mac OSX (x64)
The [Engine (macosx) build](https://github.com/enso-org/enso/actions/runs/5769482320/job/15641558873?pr=7487#step:10:13812) is failing. I've just reproduced it on my own Mac and following tests are failing:
```
sbt> runEngineDistribution --run test/Tests
    - [FAILED] should parse custom format of zoned time [11ms]
        Reason: 'UTC' did not equal 'Etc/UTC' (at /enso/test/Tests/src/Data/Time/Date_Time_Spec.enso:48:5-24).
--
    - [FAILED] should parse custom format of zoned time [5ms]
        Reason: 'UTC' did not equal 'Etc/UTC' (at /enso/test/Tests/src/Data/Time/Date_Time_Spec.enso:48:5-24).
--
    - [FAILED] should parse custom format of zoned time [2ms]
        Reason: 'UTC' did not equal 'Etc/UTC' (at /enso/test/Tests/src/Data/Time/Date_Time_Spec.enso:48:5-24).
--
    - [FAILED] should parse custom format of zoned time [4ms]
        Reason: 'UTC' did not equal 'Etc/UTC' (at /enso/test/Tests/src/Data/Time/Date_Time_Spec.enso:48:5-24).

```

The error seems to persist. Let's investigate it cause!

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Mitigates #7505 and makes sure CI passes again on our x64 based Mac OS X machines.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the `Enso` style guides. 
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: test/Tests/src/Data/Time/Date_Time_Spec.enso [hunk 1]
```diff
@@ -190,7 +190,8 @@ spec_with name create_new_datetime parse_datetime nanoseconds_loss_in_precision=
             time . millisecond . should_equal 0
             time . microsecond . should_equal 0
             time . nanosecond . should_equal 0
-            loose_zone_equal time.zone.zone_id  "Etc/UTC"
+            loose_zone_equal (time.zone.zone_id . take (Last 3) . to_case Case.Upper)  "UTC"
+            loose_zone_equal time.zone Time_Zone.utc
```
Comment (radeusgd) [inline test/Tests/src/Data/Time/Date_Time_Spec.enso]: IIRC this last line won't pass - as I posted:
```
> Date_Time.parse "2020-05-06 04:30:20 UTC" "yyyy-MM-dd HH:mm:ss z" . zone == Time_Zone.utc
>>> false
```
On Windows the `Etc/UTC` timezone is not considered equal to `UTC`.

Comment (radeusgd) [inline test/Tests/src/Data/Time/Date_Time_Spec.enso]:
```suggestion
```

**Review by radeusgd — APPROVED** [review]
If you want the check to pass, I think you need to remove the second added line, I'm pretty sure it will fail.

Definitely good to fix CI to pass, as always failing on Mac is not good - so I'm _For_ merging this PR.

However, I would like to **not** close #7505. I think we should investigate why one system returns `Etc/UTC` and other just `UTC`. 

[Apparently they seem to be different names for the same concept...](https://stackoverflow.com/questions/14128574/is-there-a-difference-between-the-utc-and-etc-utc-time-zones)

Still, I'd like us to ensure that the line I just suggested you to remove actually works in longer term:
```
loose_zone_equal time.zone Time_Zone.utc
```

As a quick fix for CI - this is fine, but for longer-term it would be worth to investigate why there is this timezone inconsistency and fix it.

As users will definitely expect the TZ returned here to `== Time_Zone.utc`.

File: test/Tests/src/Data/Time/Date_Time_Spec.enso (hunk 1 above)
Comment (radeusgd) [reply test/Tests/src/Data/Time/Date_Time_Spec.enso]: Fine to remove this line to get CI back working.

For longer term, we will probably want to ensure that this just returns a timezone that `== Time_Zone.utc`.

**Review by GregoryTravis — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
There are two more `[FAILED` tests in the Mac OS X engine CI run:
- JavascriptDate: [should handle shifting dates around spring DST edge cases](https://github.com/enso-org/enso/actions/runs/5809531874/job/15748492229?pr=7526#step:10:13571)
- JavascriptDataInArray: [should handle shifting dates around spring DST edge cases](https://github.com/enso-org/enso/actions/runs/5809531874/job/15748492229?pr=7526#step:10:13782)

On my Mac the tests are skipped, but not on CI...

**Comment by JaroslavTulach** [comment]
Having CI green is important. I guess I merge now. @mwu-tow, can you make Mac OS X CI _required_ again? Otherwise we are likely going to regress soon.

**Pull request merged**